canonical-web-and-design/jaas-dashboard
# Refactor apps panel

## Done

- Refactor apps panel to decouple from ModelDetails
- Added URL persistence through query strings
- Added loading state to slide panels
- Driveby: Short circuit `getControllerDataByUUID` if `!controllerData` 
- Driveby: add conditional checks to couple of util queries 

## QA

- Pull code
- Run `./run clean && ./run serve`
- Open http://0.0.0.0:8036/
- Click app in model details - verify correct app info appears
- Refresh the page, verify panel persists with loading state

## Details

Fixes #677, Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1415
